app/models/concerns/project_association.rb

```diff
@@ -158,8 +158,8 @@ def is_unique
 
     def set_media
       unless self.url.blank? && self.quote.blank? && self.file.blank? && self.media_type != 'Blank'
-        m = self.create_media
-        self.media_id = m.id unless m.nil?
+        self.create_media!
+        self.media_id unless self.media_id.nil?
```

Comment:
Q: this line should return `self.media_id` or assign a value to `self.media_id`

Reply:
The `create_media` method changes the object and assigns the value, if I'm not mistaken. So I think we should be able to just return `self.media_id`.